templates/messageextension/csharp/default/Bot/TeamsMessageExtension.tpl

```diff
@@ -0,0 +1,210 @@
+﻿using Microsoft.Bot.Builder;
+using Microsoft.Bot.Builder.Teams;
+using Microsoft.Bot.Schema;
+using Microsoft.Bot.Schema.Teams;
+
+using Newtonsoft.Json.Linq;
+
+namespace {{BlazorAppServer}}.Bot;
+
+public class TeamsMessageExtension : TeamsActivityHandler
+{
+    // Message Extension Code
+    // Action.
+    protected override async Task<MessagingExtensionActionResponse> OnTeamsMessagingExtensionSubmitActionAsync(ITurnContext<IInvokeActivity> turnContext, MessagingExtensionAction action, CancellationToken cancellationToken)
+    {
+        switch (action.CommandId)
+        {
+            case "createCard":
+                return CreateCardCommand(turnContext, action);
+            case "shareMessage":
+                return ShareMessageCommand(turnContext, action);
+        }
+        return new MessagingExtensionActionResponse();
+    }
+
+    private MessagingExtensionActionResponse CreateCardCommand(ITurnContext<IInvokeActivity> turnContext, MessagingExtensionAction action)
+    {
+        // The user has chosen to create a card by choosing the 'Create Card' context menu command.
+        var createCardData = ((JObject)action.Data).ToObject<CardResponse>();
+
+        var card = new HeroCard
+        {
+            Title = createCardData.Title,
+            Subtitle = createCardData.Subtitle,
+            Text = createCardData.Text,
+        };
+
+        var attachments = new List<MessagingExtensionAttachment>();
+        attachments.Add(new MessagingExtensionAttachment
+        {
+            Content = card,
+            ContentType = HeroCard.ContentType,
+            Preview = card.ToAttachment(),
+        });
+
+        return new MessagingExtensionActionResponse
+        {
+            ComposeExtension = new MessagingExtensionResult
+            {
+                AttachmentLayout = "list",
+                Type = "result",
+                Attachments = attachments,
+            },
+        };
+    }
+
+    private MessagingExtensionActionResponse ShareMessageCommand(ITurnContext<IInvokeActivity> turnContext, MessagingExtensionAction action)
```

Comment:
Same here, if we have a command handler, these 2 commands can be more simple to understand and easy to add more commands.